account_move_name_sequence/tests/test_account_move_name_seq.py

```diff
@@ -164,3 +178,379 @@ def test_in_refund(self):
+                    self._create_payment_form(env)
+                # sleep in order to avoid release the locks too faster
+                # It could be many methods called after creating these
+                # kind of records e.g. reconcile
+                _logger.info("Finishing waiting %s" % (deadlock_timeout + 12))
+                time.sleep(deadlock_timeout + 12)
+
+    def test_sequence_concurrency_10_draft_invoices(self):
+        """Creating 2 DRAFT invoices not should raises errors"""
+        with self._new_cr() as cr0, self._new_cr() as cr1, self._new_cr() as cr2:
+            env0 = api.Environment(cr0, SUPERUSER_ID, {})
+            env1 = api.Environment(cr1, SUPERUSER_ID, {})
+            env2 = api.Environment(cr2, SUPERUSER_ID, {})
+            for cr in [cr0, cr1, cr2]:
+                # Set 10s timeout in order to avoid waiting for release locks a long time
+                cr.execute("SET LOCAL statement_timeout = '10s'")
+
+            # Create "last move" to lock
+            invoice = self._create_invoice_form(env0)
+            self.addCleanup(self._clean_moves, invoice.ids)
+            env0.cr.commit()
+            try:
+                with env1.cr.savepoint(), env2.cr.savepoint():
+                    invoice1 = self._create_invoice_form(env1, post=False)
+                    self.assertEqual(invoice1.state, "draft")
+                    invoice2 = self._create_invoice_form(env2, post=False)
+                    self.assertEqual(invoice2.state, "draft")
+            except psycopg2.OperationalError as e:
+                if e.pgcode in PG_CONCURRENCY_ERRORS:
+                    self.assertFalse(
+                        True,
+                        "Should it raises error to user and rollback the whole transaction? %s"
+                        % e,
+                    )
+                else:
+                    raise
+
+    def test_sequence_concurrency_20_editing_last_invoice(self):
+        """Edit last invoice and create a new invoice
+        should not raises errors"""
+        with self._new_cr() as cr0, self._new_cr() as cr1:
+            env0 = api.Environment(cr0, SUPERUSER_ID, {})
+            env1 = api.Environment(cr1, SUPERUSER_ID, {})
+            for cr in [cr0, cr1]:
+                # Set 10s timeout in order to avoid waiting for release locks a long time
+                cr.execute("SET LOCAL statement_timeout = '10s'")
+
+            # Create "last move" to lock
+            invoice = self._create_invoice_form(env0)
+
+            self.addCleanup(self._clean_moves, invoice.ids)
+            env0.cr.commit()
+            try:
+                with env0.cr.savepoint(), env1.cr.savepoint():
+                    # Edit something in "last move"
+                    invoice.write({"write_uid": env0.uid})
+                    invoice.flush()
+                    self._create_invoice_form(env1)
+            except psycopg2.OperationalError as e:
```

Comment:
Remove all the `except` sections for this PR

It was only to proof the issue in original Odoo's PR